tests/test_models.py

```diff
@@ -0,0 +1,194 @@
+    #                                        params=params,
+    #                                        stratified=True)
+    # assert max_acc_model['accuracy'] >= 0.5
+
+
+def test_pca():
+    # unlabeled data split
+    X, Ux, y, Uy = train_test_split(pytest.spectra,
+                                    pytest.labels,
+                                    test_size=0.5,
+                                    random_state=0)
+    Uy = np.full_like(Uy, -1)
+    # data split for data visualization
+    X_train, X_test, y_train, y_test = train_test_split(X,
+                                                        y,
+                                                        test_size=0.2,
+                                                        random_state=0)
+
+    filename = 'test_pca'
+    pcs = utils.pca(X_train, Ux, 2)
+    utils.plot_pca(pcs, y_train, np.full_like(Uy, -1), filename, 2)
+    os.remove(filename+'.png')
+
+    # filename = 'test_multiD_pca'
+    # utils.multiD_pca(X_train, y_train, Ux, np.full_like(Uy, -1), filename, n=5)
+    # os.remove(filename+'.png')
+
+    # normalization
+    normalizer = StandardScaler()
+    normalizer.fit(X_train)
+
+    X_train = normalizer.transform(X_train)
+    X_test = normalizer.transform(X_test)
+
+    # default behavior
+    model = LogReg(params=None, random_state=0)
+    model.train(X_train, y_train)
+
+    # testing train and predict methods
+    pred, acc = model.predict(X_test, y_test)
+
+    filename = 'test_cf'
+    utils.plot_cf(y_test, pred, title=filename, filename=filename)
+    os.remove(filename+'.png')
+
+
+def test_LogReg():
+    # test saving model input parameters
+    # missing 'C' key
+    params = {'max_iter': 2022, 'tol': 0.5}
+    with pytest.raises(ValueError):
+        LogReg(params=params)
+    params = {'max_iter': 2022, 'tol': 0.5, 'C': 5.0}
+    model = LogReg(params=params)
+
+    assert model.model.max_iter == params['max_iter']
+    assert model.model.tol == params['tol']
+    assert model.model.C == params['C']
+
+    X_train, X_test, y_train, y_test = train_test_split(pytest.spectra,
+                                                        pytest.labels,
+                                                        test_size=0.2,
+                                                        random_state=0)
+
+    # normalization
+    normalizer = StandardScaler()
+    normalizer.fit(X_train)
+
+    X_train = normalizer.transform(X_train)
+    X_test = normalizer.transform(X_test)
+
+    # default behavior
+    model = LogReg(params=None, random_state=0)
+    model.train(X_train, y_train)
+
+    # testing train and predict methods
+    pred, acc = model.predict(X_test, y_test)
+
+    # since the test data used here is synthetic/toy data (i.e. uninteresting),
+    # the trained model should be at least better than a 50-50 guess
+    # if it was worse, something would be wrong with the ML class
+    assert acc > 0.5
```

Comment:
I added some clarification to explain our "good-enough" testing of trained ML models.